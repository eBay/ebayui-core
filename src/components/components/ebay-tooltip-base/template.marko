<span
    w-on-expander-expand="handleExpand"
    w-on-expander-collapse="handleCollapse"
    w-body
    overlay-style=data.overlayStyle
    w-bind/>
