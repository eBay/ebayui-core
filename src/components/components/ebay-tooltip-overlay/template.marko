<script marko-init>
    var constants = require("./constants");
    var typeRoles = constants.typeRoles;
    var pointerStyles = constants.pointerStyles;
</script>

<var hasUserStyles=(
    data.styleTop ||
    data.styleLeft ||
    data.styleRight ||
    data.styleBottom
)/>
<var overlayStyles=(hasUserStyles ? {
    top: data.styleTop,
    left: data.styleLeft,
    right: data.styleRight,
    bottom: data.styleBottom
} : pointerStyles[data.pointer || "bottom"])/>

<div
    class="${data.type}__overlay"
    role=typeRoles[data.type]
    aria-labelledby=(data.type === "tourtip" && widget.elId("tourtip-label"))
    id=data.tooltipId
    style=overlayStyles
    w-bind>
    <span class="${data.type}__pointer ${data.type}__pointer--${data.pointer}"/>
    <div class="${data.type}__mask">
        <div class="${data.type}__cell">
            <div class="${data.type}__content">
                <if(data.heading)>
                    <h2
                        class=["${data.type}__heading", data.heading.class]
                        style=data.heading.style
                        w-id="tourtip-label"
                        w-body=data.heading.renderBody/>
                </if>

                <if(data.content)>
                    <div
                        body-only-if(!data.content.class && !data.content.style)
                        class=data.content.class
                        style=data.content.style
                        w-body=data.content.renderBody/>
                </if>
            </div>

            <if(data.type !== "tooltip")>
                <button
                    aria-label=data.a11yCloseText
                    class="${data.type}__close"
                    type="button"
                    w-on-click="handleCloseButton">
                    <span aria-hidden="true" role="img"/>
                </button>
            </if>
        </div>
    </div>
</div>
