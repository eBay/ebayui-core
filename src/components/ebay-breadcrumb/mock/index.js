const getItem = (text, href) => {
    const item = {
        href: 'https://www.ebay.com/',
        navSrc: '{"actionKind":"NAVSRC","operationId":"2489527"}',
        _sp: 'p2489527.m4340.l9751.c1'
    };
    const clonedItem = JSON.parse(JSON.stringify(item));
    clonedItem.renderBody = (stream) => stream.write(text);
    clonedItem.href = href;
    return clonedItem;
};
const basicItems = [getItem('eBay', 'https://www.ebay.com'),
    getItem('Auto Parts and Vehicles', 'https://www.ebay.com/b/Auto-Parts-and-Vehicles/6000/bn_1865334'),
    getItem('Motors Parts & Accessories', null)];

const firstItemMissingHref = [getItem('eBay', null),
    getItem('Auto Parts and Vehicles', 'https://www.ebay.com/b/Auto-Parts-and-Vehicles/6000/bn_1865334'),
    getItem('Motors Parts & Accessories', null)];

module.exports = {
    basicItems: {
        headingText: 'Page navigation',
        items: basicItems
    },
    hijax: {
        hijax: true,
        items: basicItems
    },
    firstItemMissingHref: {
        headingText: 'Page navigation',
        items: firstItemMissingHref
    },
    itemsWithHeadingLevel: {
        headingText: 'Page navigation',
        headingLevel: 'h3',
        items: basicItems
    }
};
