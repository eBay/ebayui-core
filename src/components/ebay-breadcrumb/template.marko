<script marko-init>
    var processHtmlAttributes = require("../../common/html-attributes");
</script>

<nav w-bind
    aria-labelledby=widget.elId("breadcrumb-heading")
    class=["breadcrumb", data.class]
    style=data.style
    role="navigation"
    ${processHtmlAttributes(data)}>

    <${data.a11yHeadingTag || "h2"} id=widget.elId("breadcrumb-heading") class="clipped">
        ${data.a11yHeadingText}
    </>
    <ul>
        <li for(item in data.items | status-var="status")>
            <var current=(!item.href && status.isLast())/>
            <${item.href ? "a" : "button"}
                href=item.href
                aria-current=(current && "location")
                class=item.class
                style=item.style
                w-body=item.renderBody
                w-onclick=(!current && "handleClick")
                ${processHtmlAttributes(item)}/>
        </li>
    </ul>
</nav>
