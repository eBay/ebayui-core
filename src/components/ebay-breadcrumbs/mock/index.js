import { createRenderBody, getNItems } from '../../../common/test-utils/shared';

export const Links = {
    a11yHeadingText: 'Page navigation',
    items: getNItems(3, (i) => ({
        href: i === 2 ? undefined : '#',
        renderBody: createRenderBody(`Item Text ${i}`),
    })),
};
export const Buttons = {
    items: getNItems(3, (i) => ({
        renderBody: createRenderBody(`Item Text ${i}`),
    })),
};
export const linkLastWithoutHREF = {
    a11yHeadingText: 'Page navigation',
    items: getNItems(3, (i) => ({
        href: i === 2 ? undefined : '#',
        renderBody: createRenderBody(`Item Text ${i}`),
    })),
};
export const linkHeadingTag = Object.assign({}, Links, {
    a11yHeadingTag: 'h3',
});
