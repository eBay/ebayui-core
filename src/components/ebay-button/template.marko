<script marko-init>
    var getWidgetId = require("../../common/get-marko-3-widget-id");
    var processHtmlAttributes = require("../../common/html-attributes");
</script>

<var size=(data.size || "medium")/>
<var priority=(data.priority || "secondary")/>
<var variant=(!data.variant && data.href ? "fake" : data.variant)/>
<var isIconVariant=(variant === 'icon')/>
<var isExpandVariant=(variant === 'expand')/>
<var isCtaVariant=(variant === 'cta')/>
<var isBadged=Boolean(data.badgeNumber && isIconVariant)/>
<var noText=(isIconVariant || isBadged || (isExpandVariant && data.noText))/>
<var baseClass=(variant ? "${variant}-btn" : "btn")/>
<var sizeClass="${baseClass}--${size}"/>
<var htmlAttributes=processHtmlAttributes(data)/>
<var tag=(data.href ? "a" : "button")/>

<${tag}
    ${htmlAttributes}
    w-bind
    w-onclick="handleClick"
    w-onkeydown="handleKeydown"
    id=(data.id || getWidgetId(out))
    class=[
        data.class,
        baseClass,
        noText && "${baseClass}--no-text",
        isBadged && "${baseClass}--badged",
        data.fluid && "${baseClass}--fluid",
        data.truncate && "${sizeClass}-truncated",
        data.fixedHeight && "${sizeClass}-fixed-height",
        !data.fixedHeight && !data.truncate && sizeClass,
        (priority === "secondary" || priority === "primary") && "${baseClass}--${priority}"
    ]
    style=data.style
    href=data.href
    type=(tag === "button" && data.type || "button")
    disabled=data.disabled
    aria-disabled=(data.partiallyDisabled && "true")>

    <if(isCtaVariant)>
        <span class="cta-btn__cell">
            <span w-body/>
            <ebay-icon type="inline" name="arrow-right-bold" class="cta-btn__icon" no-skin-classes/>
        </span>
    </if>
    <else>
        <var hasAriaLabel=Boolean(htmlAttributes.ariaLabel)/>
        <span body-only-if(!hasAriaLabel) w-body aria-hidden="true"/>
        <if(isBadged)>
            <ebay-badge
                number=data.badgeNumber
                type="icon"
                aria-label=(hasAriaLabel && data.badgeAriaLabel)
                aria-hidden=(hasAriaLabel && "true")/>
        </if>
    </else>
</>
