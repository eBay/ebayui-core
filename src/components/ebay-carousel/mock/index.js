const renderBody = (stream) => stream.write('text');
const itemWidth = 200;
const item = { renderBody, '*': { style: `width:${itemWidth}px` } };
const threeItems = [item, item, item];
const sixItems = threeItems.concat(threeItems);
const twelveItems = sixItems.concat(sixItems);

// mocks for visual debugging
const debugItem = { renderBody, '*': { style: `height:200px;width:400px;background:gray` } };
const debugItems = [debugItem, debugItem, debugItem, debugItem, debugItem, debugItem];

module.exports = { renderBody, itemWidth, item, threeItems, sixItems, twelveItems, debugItems };
