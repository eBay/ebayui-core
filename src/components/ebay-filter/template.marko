<script marko-init>
    var processHtmlAttributes = require("../../common/html-attributes");
</script>

<var classPrefix=(data.href ? "filter-link" : "filter-button")/>

<${data.href ? "a" : "button"}
    w-bind
    w-on-click="handleButtonClick"
    class=[
        classPrefix,
        data.pressed ? classPrefix + "--selected" : classPrefix + "--unselected",
        data.class
    ]
    style=data.style
    href=data.href
    disabled=data.disabled
    type=(!data.href && "button")
    aria-pressed=(!data.href && data.pressed && "true")
    ${processHtmlAttributes(data)}>
    <span class="${classPrefix}__cell">
        <span w-body/>
        <if(data.href && data.pressed)>
            <span w-id="active-text" class="clipped"> - ${data.a11yActiveText || "Selected"}</span>
        </if>
    </span>
</>
