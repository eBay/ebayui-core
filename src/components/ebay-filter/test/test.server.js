import { describe, it, expect } from "vitest";

import { render } from "@marko/testing-library";
import template from "../index.marko";
import { testPassThroughAttributes } from "../../../common/test-utils/server";
import * as mock from "./mock";

describe("filter", () => {
    it("renders defaults", async () => {
        const input = mock.Basic;
        const { getByRole } = await render(template, input);
        const filterEl = getByRole("button");
        expect(filterEl).toMatchSnapshot();
    });

    it("renders with pressed attribute", async () => {
        const input = mock.Selected;
        const { getByRole } = await render(template, input);
        expect(getByRole("button")).toMatchSnapshot();
    });

    it("renders without pressed attribute", async () => {
        const input = mock.withOutPressed;
        const { getByRole } = await render(template, input);
        expect(getByRole("button")).toMatchSnapshot();
    });

    it("renders with disabled attribute", async () => {
        const input = mock.Disabled;
        const { getByRole } = await render(template, input);
        expect(getByRole("button")).toMatchSnapshot();
    });

    it("renders fake version", async () => {
        const input = mock.Fake;
        const { getByText } = await render(template, input);
        const filterTextEl = getByText(input.renderBody.text);
        const filterEl = filterTextEl.closest("a");
        expect(filterEl).toMatchSnapshot();
    });

    it("renders fake version with disabled attribute", async () => {
        const input = mock.fakeDisabled;
        const { getByText } = await render(template, input);
        expect(getByText(input.renderBody.text).closest("a")).toMatchSnapshot();
    });

    it("renders fake version with pressed attribute", async () => {
        const input = mock.fakeSelected;
        const { getByText } = await render(template, input);
        expect(getByText(input.renderBody.text).closest("a")).toMatchSnapshot();
    });

    testPassThroughAttributes(template, {
        getClassAndStyleEl(component) {
            return component.getByRole("button");
        },
    });
});
