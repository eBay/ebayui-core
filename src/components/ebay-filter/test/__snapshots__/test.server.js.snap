// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`filter > passes through additional html attributes 1`] = `
<button
  class="filter-button filter-button--unselected class1"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  style="color:red"
  type="button"
>
  <span
    class="filter-button__cell"
  >
    <span />
  </span>
</button>
`;

exports[`filter > passes through additional html attributes 2`] = `
<button
  class="filter-button filter-button--unselected class1"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  style="color:red"
  type="button"
>
  <span
    class="filter-button__cell"
  >
    <span />
  </span>
</button>
`;

exports[`filter > renders defaults 1`] = `
<button
  class="filter-button filter-button--unselected"
  type="button"
>
  <span
    class="filter-button__cell"
  >
    <span>
      <!--F#3-->
      text
      <!--F/-->
    </span>
  </span>
</button>
`;

exports[`filter > renders fake version 1`] = `
<a
  class="filter-link filter-link--unselected"
  href="#fake"
>
  <span
    class="filter-link__cell"
  >
    <span>
      <!--F#3-->
      text
      <!--F/-->
    </span>
  </span>
</a>
`;

exports[`filter > renders fake version with disabled attribute 1`] = `
<a
  class="filter-link filter-link--unselected"
  disabled=""
  href="#fake"
>
  <span
    class="filter-link__cell"
  >
    <span>
      <!--F#3-->
      text
      <!--F/-->
    </span>
  </span>
</a>
`;

exports[`filter > renders fake version with pressed attribute 1`] = `
<a
  class="filter-link filter-link--selected"
  href="#fake"
>
  <span
    class="filter-link__cell"
  >
    <span>
      <!--F#3-->
      text
      <!--F/-->
    </span>
    <span
      class="clipped"
      id="s0-active-text"
    >
      - Selected Filter
    </span>
  </span>
</a>
`;

exports[`filter > renders with disabled attribute 1`] = `
<button
  class="filter-button filter-button--unselected"
  disabled=""
  type="button"
>
  <span
    class="filter-button__cell"
  >
    <span>
      <!--F#3-->
      text
      <!--F/-->
    </span>
  </span>
</button>
`;

exports[`filter > renders with pressed attribute 1`] = `
<button
  aria-pressed="true"
  class="filter-button filter-button--selected"
  type="button"
>
  <span
    class="filter-button__cell"
  >
    <span>
      <!--F#3-->
      text
      <!--F/-->
    </span>
  </span>
</button>
`;

exports[`filter > renders without pressed attribute 1`] = `
<button
  class="filter-button filter-button--selected"
  type="button"
>
  <span
    class="filter-button__cell"
  >
    <span>
      <!--F#3-->
      text
      <!--F/-->
    </span>
  </span>
</button>
`;
